framework/src/meshgenerators/TriangleMesher.C

```diff
@@ -0,0 +1,169 @@
+    std::unique_ptr<ReplicatedMesh> rmesh = dynamic_pointer_cast<ReplicatedMesh>(*mesh);
+    auto bpoints = rmesh->get_boundary_points();
+    for (auto & part : bpoints)
+    {
+      std::vector<unsigned int> seg_indices(1, 0);
+      for (auto & points : part.second)
+      {
+        seg_indices.push_back(seg_indices.back() + points.size());
+        for (auto & p : points)
+          boundary_points[part.first].push_back(p);
+      }
+
+      TriangleInterface::Hole * hole = new TriangleInterface::ArbitraryHole(
+          part.first, boundary_points[part.first], seg_indices);
+      _holes.push_back(hole);
+    }
+  }
+
+  // Clear out any data which may have been in the Mesh
+  mesh->clear();
+
+  // Make sure the new Mesh will be 2D
+  mesh->set_mesh_dimension(2);
+
+  // process input parameters
+  auto & points = getParam<std::vector<Real>>("enclosing_points");
+  unsigned int npoints = points.size() / 2;
+  if (npoints * 2 != points.size())
+    mooseError("Number of values in 'points' must be even");
+
+  std::vector<unsigned int> div(npoints, 1);
+  if (isParamValid("segment_subdivisions"))
+    div = getParam<std::vector<unsigned int>>("segment_subdivisions");
+
+  std::vector<int> raw_markers;
+  if (isParamValid("segment_markers"))
+  {
+    raw_markers = getParam<std::vector<int>>("segment_markers");
+    if (raw_markers.size() != npoints)
+      mooseError("Number of boundary markers is not equal to number of boundary segments");
+    for (auto & marker : raw_markers)
+      if (marker <= 1)
+        mooseError("Markers need to be strictly greater than 1");
+  }
+  else
+    raw_markers.resize(npoints, 2);
+
+  // add points and process boundary markers
+  std::vector<int> markers;
+  for (unsigned int p = 0; p < npoints; ++p)
+  {
+    Point x0(points[p + p], points[p + p + 1]);
+    Point x1;
+    if (p == npoints - 1)
+      x1 = Point(points[0], points[1]);
+    else
+      x1 = Point(points[p + p + 2], points[p + p + 3]);
+    for (unsigned int i = 0; i < div[p]; ++i)
+    {
+      mesh->add_point(x0 + i * (x1 - x0) / div[p]);
+      markers.push_back(raw_markers[p]);
+    }
+  }
+
+  // Construct the Triangle Interface object
+  // Note: we are using the implicit segment from the enclosing points.
+  TriangleInterface t(dynamic_cast<UnstructuredMesh &>(*mesh));
+
+  // Set custom variables for the triangulation
+  t.desired_area() = getParam<Real>("max_area");
+  t.minimum_angle() = getParam<Real>("min_angle");
+  t.triangulation_type() = TriangleInterface::PSLG;
+  t.elem_type() = TRI3;
+  // use Y command line switch to prohibit the insertion of Steiner points on the mesh enclosure
+  // including the enclosures of all holes
+  t.extra_flags() = "Y";
+
+  t.attach_boundary_marker(&markers);
+
+  if (_holes.size() > 0)
+    t.attach_hole_list(&_holes);
+
+  // Triangulate!
+  t.triangulate();
+
+#else
+  mooseError("Triangle is not available in libMesh");
```

Comment:
Can we tell the user how to configure here? It might head off a question or two on the mailing list. I'll do a more thorough review when I get caught up. Won't this fail on our test boxes? We might have to add new options in the TestHarness for these new configuration (like we do for SuperLU, MUMPS, etc).

Reply:
Yes, need to add `--disable-strict-lgpl` here.